pkg/admission/validator/lakom/shoot.go

```diff
@@ -0,0 +1,85 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package cache
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gardener/gardener-extension-shoot-lakom-service/pkg/apis/lakom"
+	extensionswebhook "github.com/gardener/gardener/extensions/pkg/webhook"
+	"github.com/gardener/gardener/pkg/apis/core"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// shoot validates shoots
+type shoot struct {
+	apiReader client.Reader
+	decoder   runtime.Decoder
+}
+
+// NewShootValidator returns a new instance of a shoot validator.
+func NewShootValidator(apiReader client.Reader, decoder runtime.Decoder) extensionswebhook.Validator {
+	return &shoot{
+		apiReader: apiReader,
+		decoder:   decoder,
+	}
+}
+
+// TODO: Some func in gardener/gardener instead of this? This was pulled from registry-cache extension
+// FindExtension finds extension with the given type.
+// The first return argument is index of the extension in the list. -1 is returned if the extension is not found.
+// The second return argument is the extension itself. An empty extension is returned if the extension is not found.
+func findExtension(extensions []core.Extension, extensionType string) (int, core.Extension) {
+	for i, ext := range extensions {
+		if ext.Type == extensionType {
+			return i, ext
+		}
+	}
+
+	return -1, core.Extension{}
+}
```

Comment:
Could I use some util from gardener/gardener? Seems like this should be a common task for all extensions that have a providerConfig in the shoot